solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCancelMessageHandler.java

```diff
@@ -0,0 +1,237 @@
+
+    NamedList<Object> results = new SimpleOrderedMap<>();
+
+    try {
+      boolean isInProgress = overseerTaskProcessor.isAsyncTaskInProgress(targetAsyncId);
+      boolean isSubmitted = overseerTaskProcessor.isAsyncTaskInSubmitted(targetAsyncId);
+
+      if (!isSubmitted && !isInProgress) {
+        results =
+            buildResponse(
+                targetAsyncId, "not_found", "Task was not found or has already completed.");
+        return new OverseerSolrResponse(results);
+      }
+
+      if (!isInProgress) {
+        // Task is not in progress; attempt to remove from the queue
+        boolean removed = overseerTaskProcessor.removeSubmittedTask(targetAsyncId);
+        if (removed) {
+          results =
+              buildResponse(
+                  targetAsyncId,
+                  "removed",
+                  "Submitted Task was successfully removed from the queue.");
+        } else {
+          results =
+              buildResponse(
+                  targetAsyncId, "not_found", "Task was not found or has already completed.");
+        }
+        return new OverseerSolrResponse(results);
+      }
+
+      // Task is in progress
+      return handleTaskCancellation(targetAsyncId, results);
+
+    } catch (Exception e) {
+      log.error("Error cancelling task {}", targetAsyncId, e);
+      results.add(
+          "response",
+          buildResponse(
+              targetAsyncId,
+              "error",
+              "Exception occurred while cancelling task: " + e.getMessage()));
+      return new OverseerSolrResponse(results);
+    }
+  }
+
+  private OverseerSolrResponse handleTaskCancellation(
+      String targetAsyncId, NamedList<Object> results)
+      throws InterruptedException, KeeperException, IOException {
+
+    // Attempt to cancel the in-progress task
+    boolean cancelled = overseerTaskProcessor.cancelInProgressAsyncTask(targetAsyncId);
+
+    if (cancelled) {
+      results = buildResponse(targetAsyncId, "cancelled", "Task was successfully cancelled.");
+    } else {
+      results =
+          buildResponse(
+              targetAsyncId, "failed", "Task could not be cancelled or has already completed.");
+    }
+
+    return new OverseerSolrResponse(results);
+  }
+
+  private NamedList<Object> buildResponse(String asyncId, String status, String message) {
+    NamedList<Object> response = new SimpleOrderedMap<>();
+    response.add("asyncId", asyncId);
+    response.add("status", status);
+    response.add("message", message);
+    return response;
+  }
+
+  @Override
+  public String getName() {
+    return "Overseer Cancel Message Handler";
+  }
+
+  @Override
+  public String getTimerName(String operation) {
+    return "OverseerCancelMessageHandlerTimer" + operation;
+  }
+
+  // TODO: find a better way to lock, currently lock based on asyncId being unique
+  public Lock lockTask(ZkNodeProps message, long ignored) {
```

Comment:
```suggestion
  @Override
  public Lock lockTask(ZkNodeProps message, long ignored) {
```

based on precommit CI feedback:

```
/home/runner/work/solr/solr/solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCancelMessageHandler.java:191: warning: [MissingOverride] lockTask implements method in OverseerMessageHandler; expected @Override
```